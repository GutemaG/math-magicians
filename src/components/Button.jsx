import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { value, handleClick, className } = props;
  return (
    <button type="button" className={className} onClick={handleClick}>
      {value}
    </button>
  );
};

Button.propTypes = {
  value: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

Button.defaultProps = {
  className: 'btn',
};

export default Button;
