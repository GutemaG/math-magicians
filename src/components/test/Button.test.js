import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Testing button rendered correctly', () => {
  it('Button rendered', () => {
    const tree = renderer.create(
      <Button value="AC" handleClick={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
