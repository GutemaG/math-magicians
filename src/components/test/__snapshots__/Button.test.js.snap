// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button rendered correctly Button rendered 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
